packages/js/src/executors/release-publish/extract-npm-publish-json-data.ts

```diff
@@ -0,0 +1,62 @@
+const expectedNpmPublishJsonKeys = [
+  'id',
+  'name',
+  'version',
+  'size',
+  'filename',
+];
+
+export function extractNpmPublishJsonData(str: string): {
+  beforeJsonData: string;
+  jsonData: Record<string, unknown> | null;
+  afterJsonData: string;
+} {
+  const jsonRegex = /{(?:[^{}]|{[^{}]*})*}/g;
```

Comment:
Could you leave a comment with exactly what this regex matches?

Does it match nested JSON objects such as `{ "key": "value" }` in `{ "a": { "key": "value" } }`.

Reply:
Added a detailed explainer comment